orchestrator/orchestrator.go

```diff
@@ -223,3 +260,42 @@ func (s *Orchestrator) processTask(nodeKey string, task *process.Process_Node_Ta
 	}, s.accountName, s.accountPassword)
 	return err
 }
+
+func query(data *types.Struct, path *process.Process_Node_Map_Output_Reference_Path) (*types.Value, error) {
+	var v *types.Value
+	key, ok := path.Path.Selector.(*process.Process_Node_Map_Output_Reference_Path_Key)
+	if !ok {
+		return nil, fmt.Errorf("orchestrator: first selector in the path must be always key")
+	}
+
+	v, ok = data.Fields[key.Key]
+	if !ok {
+		return nil, fmt.Errorf("orchestrator: key %s not found", key.Key)
+	}
+
+	for p := path.Path; p.Path != nil; p = p.Path {
+		switch s := p.Selector.(type) {
+		case *process.Process_Node_Map_Output_Reference_Path_Key:
```

Comment:
you could transform this case to a simple if / else if / else.
At least, add a `default` that return an error.